src/include/neorados/RADOS.hpp

```diff
@@ -407,6 +407,24 @@ class Op {
   }
  public:
 
+  /**
+   * Execute an OSD class method on an object
+   *
+   * The OSD has a plugin mechanism for performing complicated
+   * operations on an object atomically. These plugins are called
+   * classes. This function allows librados users to call the custom
+   * methods. The input and output formats are defined by the class.
+   * Classes in ceph.git can be found in src/cls subdirectories
+   *
+   * This Op may be a read, so only read execs are permitted with this interface
+   *
+   * @param method the method as defined in cls/<class>/cls_<class>_ops.h
```

Comment:
This basically says that a _Ceph Class_ should exist within the Ceph repo. I'm not going to disagree ;-).